File: sdks/java/basys.sdk/src/org/eclipse/basyx/aas/backend/connected/aas/submodelelement/property/ConnectedSingleProperty.java
package org.eclipse.basyx.aas.backend.connected.aas.submodelelement.property;

import org.eclipse.basyx.aas.api.exception.ServerException;
import org.eclipse.basyx.aas.api.resources.ISingleProperty;
import org.eclipse.basyx.aas.api.resources.PropertyType;
import org.eclipse.basyx.vab.core.proxy.VABElementProxy;

/**
 * Connects to a PropertySingleValued as specified by DAAS containing a simple
 * value
 * 
 * @author schnicke
 *
 */
public class ConnectedSingleProperty extends ConnectedProperty implements ISingleProperty {

	public ConnectedSingleProperty(String path, VABElementProxy proxy) {
		super(PropertyType.Single, path, proxy);
	}

	@Override
	public Object get() throws Exception {
		return getProxy().readElementValue(constructPath("value"));
	}

	@Override
	public void set(Object newValue) throws ServerException {
		try {
			getProxy().updateElementValue(constructPath("value"), newValue);
		} catch (Exception e) {
			e.printStackTrace();
		}
	}

	@Override
	public void moveTo(ISingleProperty propertyName) throws ServerException {
		// TODO Auto-generated method stub
	}

	@Override
	public void setValue(Object obj) {
		// TODO Auto-generated method stub
		
	}

	@Override
	public Object getValue() {
		// TODO Auto-generated method stub
		return null;
	}

	@Override
	public void setValueId(Object obj) {
		// TODO Auto-generated method stub
		
	}

	@Override
	public Object getValueId() {
		// TODO Auto-generated method stub
		return null;
	}



}

File: sdks/c++/basys.sdk.cc/src/submodel/submodel/connected/submodelelement/property/ConnectedSingleProperty.cpp
/*
 * ConnectedSingleProperty.cpp
 *
 *      Author: wendel
 */

#include <BaSyx/submodel/connected/submodelelement/property/ConnectedSingleProperty.h>
#include <BaSyx/submodel/api/submodelelement/property/IProperty.h>

namespace basyx {
namespace submodel {

ConnectedSingleProperty::ConnectedSingleProperty(std::shared_ptr<vab::core::proxy::IVABElementProxy> proxy) :
  ConnectedProperty(PropertyType::Single, proxy)
{}

basyx::object ConnectedSingleProperty::get() const
{
  return this->retrieveObject();
}

void ConnectedSingleProperty::set(const basyx::object & value)
{
  this->setProxyValue(IProperty::Path::Value, value);
}

std::string ConnectedSingleProperty::getValueType() const
{
  auto map = this->getProxy()->readElementValue("").Get<basyx::object::object_map_t>();
  return map.at(IProperty::Path::ValueType).Get<std::string>();
}

void basyx::submodel::ConnectedSingleProperty::setValueId(const std::string & valueId)
{}

std::string basyx::submodel::ConnectedSingleProperty::getValueId() const
{
  return std::string();
}

IProperty::PropertyType basyx::submodel::ConnectedSingleProperty::getPropertyType() const
{
  return PropertyType();
}


}
}
